compiler/executable/match_/planner/plan.rs

```diff
@@ -1417,7 +1479,7 @@ impl ConjunctionPlan<'_> {
         match &self.graph.elements()[&VertexId::Pattern(pattern)] {
             PlannerVertex::Variable(_) => unreachable!("encountered variable @ pattern id {pattern:?}"),
             PlannerVertex::FunctionCall(_) => {
-                unreachable!("variable assigned to from functions cannot be produced by other instructions")
+                todo!()
```

Comment:
@dmitrii-ubskii , What does it mean if we reach this point? That we need to evaluate the function, assign, and check if they're identical to what exists?

Reply:
That means the plan is invalid. In a conjunction
```
$x isa person;
let $x = foo();
```
We expect the ordering to be:
```
FunctionCall(assign = [$x], fun = "foo")
Check(Isa($x, person))
```

Reply:
Over discord, I motivated allowing a check step for the following case:
```
$x has name "John";
($x, $y) isa friend;
($y, $z) isa pet-ownership;
let $y in hates($z);
```